Shopify/cli
# Remove go binary

### WHY are these changes introduced?
Remove all the go code for extensions. Since we want to run the last PR (that makes node the default) for 2 weeks, this PR should only be merged AFTER next weeks release.

### WHAT is this pull request doing?
I tried to do this fairly methodically with clear commit messages.  So the best way to understand the work is to look at the commit messages, but I'll try to summarize here.  In this PR we remove:

1. Code to build the dev console into the go package (we build it into the app package instead)
2. All the run time code that calls either `ui-extension-go-cli` or `ui-extensions-cli` code.
3. Remove the yaml utilities since we no longer need to deal with yaml 🎉 
4. Remove all the build, CI, or config code relating to `ui-extension-go-cli` or `ui-extensions-cli`.  I don't have much knowledge in this area, so I expect I've missed something.
5. Remove the `ui-extension-go-cli` or `ui-extensions-cli` packages.

### How to test your change?
Testing steps:

1. Run `dev && dev up`. Does it work?
2. `yarn && yarn build`
3. `cd fixtures/app`
4. Run `yarn dev`.  
5. Complete OAuth? (This shouldn't be affected but just in case)
6. Open the dev console (E.g: http://[ngrokurl]/extensions/dev-console). Show, hide and preview extensions.  Does it work?
7. `yarn generate extension`.  Now run `yarn dev`, does thew newly created extension work?
8. Run `dev && dev up`.
9. Look at the CI steps.  Do they look correct? Have I removed anything I shouldn't have?

Now look for changes that I've missed.  This is a hard problem, since finding dead code can be tricky, but here are some ideas:

* Does the deleted code import any code that is not imported anywhere else?
* Have I missed any CI code that should also be deleted?

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
